src/controls/UserEditInteraction.js

```diff
@@ -0,0 +1,218 @@
+import VectorSource from 'ol/source/Vector.js';
+import Draw from 'ol/interaction/Draw.js';
+import Modify from 'ol/interaction/Modify.js';
+import { defaultStyles } from '../constants.js';
+import {
+  getFeatureId,
+  getOlLayerForMbLayer,
+  isPointFeature,
+} from '../utils.js';
+
+/**
+ * @typedef {Object} Options
+ * @property {import('../YioMap.js').YioMap} yioMap
+ * @property {import('ol/Map.js').default} map ol map
+ */
+
+export default class UserSelectInteraction extends Interaction {
+  /** @type {import('../YioMap.js').YioMap} */
+  #yioMap = null;
+
+  /** @type {import('ol/Map.js').default} */
+  #map = null;
+
+  /**
+   * @type {VectorLayer} layer for temporary features of the draw / modify interactions
+   */
+  #editLayer = null;
+
+  getEditLayer() {
+    return this.#editLayer;
+  }
+
+  /**
+   * ol layer connected associated with the given mapbox `source-layer` (editLayer)
+   * @type {VectorLayer | import("ol/layer/VectorTile.js").default}
+   */
+  #olSourceLayer;
+
+  /**
+   * IDs of features that were modified
+   * @type {Array<string | number>}
+   */
+  #modifiedFeatureIDs = [];
+
+  /**
+   * @param {Options} options
+   */
+  constructor(options) {
+    super();
+    this.#yioMap = options.yioMap;
+    this.#map = options.map;
+
+    this.#editLayer = new VectorLayer({
+      source: new VectorSource({
+        features: [],
+      }),
+    });
+
+    this.drawInteraction = new Draw({
+      source: this.#editLayer.getSource(),
+      condition: event => {
+        const existingFeature = event.map
+          .getFeaturesAtPixel(event.pixel, {
+            hitTolerance: 7,
+          })
+          .find(f => {
+            return f.get('layer') === this.#yioMap.editLayer;
+          });
+        return !existingFeature;
+      },
+      type: 'Point',
+    });
+    this.drawInteraction.on('drawend', event => {
+      const feature = event.feature;
+      feature.set('layer', this.#yioMap.editLayer);
+      this.#editLayer.getSource().once('addfeature', () => {
+        this.#handleClick(event);
+      });
+    });
+    this.modifyInteraction = new Modify({
+      source: this.#editLayer.getSource(),
+    });
+
+    this.modifyInteraction.on('modifystart', event => {
+      const modifiedFeatureIds = event.features
+        .getArray()
+        .map(feature => getFeatureId(feature))
+        .filter(id => !!id);
+      if (modifiedFeatureIds.length) {
+        this.#modifiedFeatureIDs.push(...modifiedFeatureIds);
+        this.#olSourceLayer.changed();
+      }
+    });
+    this.modifyInteraction.on('modifyend', event => {
+      this.#handleClick(event);
+    });
+    this.drawInteraction.setActive(false);
+    this.modifyInteraction.setActive(false);
+  }
+
+  setMap(map) {
+    const currentMap = this.getMap();
+    if (currentMap) {
+      currentMap.removeLayer(this.#editLayer);
+    }
+    if (!map) {
+      return;
+    }
+    map.addLayer(this.#editLayer);
```

Comment:
This puts the `#editLayer` in a random position in the layer stack. I think it makes sense to use `#editLayer.setMap()` to get an unmanaged layer on top of other layers.